File: src/main/Java/StatementHandler.java
import org.eclipse.rdf4j.model.Statement;
import org.eclipse.rdf4j.rio.RDFHandler;
import org.eclipse.rdf4j.rio.RDFHandlerException;

import java.util.HashSet;
import java.util.Set;

public class StatementHandler implements RDFHandler {
    private Set<Statement> statements;
    public StatementHandler() {
        statements = new HashSet<>();
    }

    @Override
    public void startRDF() throws RDFHandlerException {

    }

    @Override
    public void endRDF() throws RDFHandlerException {

    }

    @Override
    public void handleNamespace(String prefix, String uri) throws RDFHandlerException {

    }

    @Override
    public void handleStatement(Statement st) throws RDFHandlerException {
        statements.add(st);
    }

    @Override
    public void handleComment(String comment) throws RDFHandlerException {

    }
}

File: src/main/scala/etl/rdf4jrefactor/StatementHandler.scala
package etl.rdf4jrefactor

import java.util

import org.eclipse.rdf4j.model.Statement
import org.eclipse.rdf4j.rio.{RDFHandler, RDFHandlerException}

import scala.collection.mutable


class StatementHandler() extends RDFHandler {
  private val statements = new mutable.HashSet[Statement]
  private val resourceProps = new mutable.HashMap[String, mutable.Map[String, AnyRef]]
  private val resourceLabels = new mutable.HashMap[String, mutable.Set[String]]
  @throws[RDFHandlerException]
  override def startRDF(): Unit = {
  }

  @throws[RDFHandlerException]
  override def endRDF(): Unit = {
  }

  @throws[RDFHandlerException]
  override def handleNamespace(prefix: String, uri: String): Unit = {
  }

  @throws[RDFHandlerException]
  override def handleStatement(st: Statement): Unit = {
    val predicate = st.getPredicate
    val subject = st.getSubject
    //includes blank nodes
    val `object` = st.getObject

    statements.add(st)
  }

  @throws[RDFHandlerException]
  override def handleComment(comment: String): Unit = {
  }

  def getStatement: mutable.HashSet[Statement] = statements
}
